packages/ra-core/src/reducer/admin/references/oneToMany.ts

```diff
@@ -13,12 +15,24 @@ interface State {
 
 type ActionTypes =
     | CrudGetManyReferenceSuccessAction
+    | CrudDeleteSuccessAction
     | { type: 'OTHER_ACTION'; payload: any; meta?: any };
 
 const oneToManyReducer: Reducer<State> = (
     previousState = initialState,
     action: ActionTypes
 ) => {
+    if (action.meta && action.meta.fetch === DELETE && action.meta.optimistic) {
```

Comment:
I suggest you group the optimistic cases just like in the resource.list.ids reducer